source/getting-started/connection.txt

```diff
@@ -0,0 +1,412 @@
+different connection mechanisms and configure the connection settings.
+
+When you connect to a MongoDB server, you provide your credentials and
+specific connection mechanism to the driver. The driver negotiates with the
+server to verify your identity, and the server grants you permissions
+appropriate to your user. If you need to set up a new user or user 
+permissions on your MongoDB server, see the **Authentication** and
+ **Authorization** sections in the 
+ `Security <https://docs.mongodb.com/manual/security/>`_ server manual.
+
+The Node.js driver can connect to MongoDB using any of the `authentication
+mechanisms 
+<https://docs.mongodb.com/manual/core/authentication/#authentication-mechanisms>`_
+supported by either MongoDB Community or Enterprise edition which includes:
+
+- ``DEFAULT``
+- ``SCRAM-SHA-256``
+- ``SCRAM-SHA-1``
+- ``MONGODB-CR``
+- ``X.509``
+- ``Kerberos (GSSAPI/SSPI)`` *(MongoDB Enterprise only)*
+- ``LDAP (PLAIN)`` *(MongoDB Enterprise only)*
+
+Connection Mechanisms
+---------------------
+
+In this section, you can find descriptions and sample code for connecting to
+a MongoDB server using each of the connection mechanism settings.
+
+DEFAULT
+~~~~~~~
+
+Specify the ``DEFAULT`` option to the driver in order to authenticate with the
+server using a username and password. This option is a fallback setting that 
+instructs the driver to negotiate the first authentication mechanism supported
+by the server using the following order of preference:
+
+#. ``SCRAM-SHA-256``
+#. ``SCRAM-SHA-1``
+#. ``MONGODB-CR``
+
+If the ``DEFAULT`` option is specified, the driver first attempts to 
+authenticate using ``SCRAM-SHA-256``. If the server does not support that 
+mechanism, the driver attempts to authenticate using ``SCRAM-SHA-1``. If the 
+server does not support that mechanism either, the driver attempts to 
+authenticate using ``MONGODB-CR``.  
+
+You can specify this option by omitting the ``authMechanism`` parameter of the
+`URI ConnectionString
+<https://docs.mongodb.com/manual/reference/connection-string/>`_ or by setting
+it explicitly to the value ``DEFAULT`` as shown in the following sample code.
+
+.. important::
+   Always **URI encode** the username and password using ``encodeURIComponent``
+   to ensure they are correctly parsed.
+   
+.. code-block:: js
+
+   const { MongoClient } = require('mongodb');
+
+   // specify the placeholder values for your environment in the following line
+   const usernanme = encodeURIComponent(<username>);
+   const password = encodeURIComponent(<password>);
+   const clusterUrl = <cluster url>;
+   const authMechanism = 'DEFAULT';
+
+   // Connection URI
+   const uri = 
+     `mongodb+srv://${username}:${password}@${clusterUrl}/?authMechanism=${authMechanism}`;
+
+   const client = new MongoClient(url);
+
+   async function run() {
+     try {
+       // Connect the client to the server
+       await client.connect();
```

Comment:
So, the node driver has anomalous behavior that isn't spec compliant like the other drivers. I suggest adding in a `ping` command, which other drivers will need to do to trigger going from a lazy to active connection.